Gordon360/Controllers/ProfilesController.cs

```diff
@@ -438,6 +439,22 @@ public async Task<ActionResult<StudentProfileViewModel>> UpdateMobilePhoneNumber
             return Ok(result);
         }
 
+        /// <summary>
+        /// Update office location (building description and room number)
+        /// </summary>
+        /// <param name="value">office location</param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("office_location/{value}")]
+        [AllowAnonymous]
+        public async Task<ActionResult<FacultyStaffProfileViewModel>> UpdateOfficeLocation(string value)
+        {
+            var username = AuthUtils.GetUsername(User);
+            var values = value.Split(',');
+            var result = await _profileService.UpdateOfficeLocationAsync(username, values[0], values[1]);
+            return Ok(result);
+        }
+
```

Comment:
```
        /// <summary>
        /// Update office location (building description and room number)
        /// </summary>
        /// <param name="value">office location</param>
        /// <returns></returns>
        [HttpPut]
        [Route("office_location/{value}")]
        [AllowAnonymous]
        public async Task<ActionResult<FacultyStaffProfileViewModel>> UpdateOfficeLocation(string value)
        {
            var username = AuthUtils.GetUsername(User);
            var values = value.Split(',');
            var result = await _profileService.UpdateOfficeLocationAsync(username, values[0], values[1]);
            return Ok(result);
        }

```

A few things here:
1) We're implementing a new `Patch` protocol that handles purely resource updating (read more on the difference between http patch & put if you want to get a better understanding of what is happening.) So your route definitely fits more under the patch protocol
2) `value` should not be passed in from the UI as a single string containing all this information. Split this into a viewmodel with 3 elements (look at any of the `recim` viewmodels with the keyword `Patch` in them. Create a viewmodel of follows a proper naming scheme

Reply:
Just turned into a patch route and created a new viewmodel